frontend/test/metabase/scenarios/admin/settings/public-sharing.cy.spec.js

```diff
@@ -0,0 +1,236 @@
+import { restore, modal, enableActionsForDB } from "__support__/e2e/helpers";
+import { SAMPLE_DB_ID } from "__support__/e2e/cypress_data";
+import { SAMPLE_DATABASE } from "__support__/e2e/cypress_sample_database";
+
+const { ORDERS_ID } = SAMPLE_DATABASE;
+
+describe("scenarios > admin > settings > public sharing", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  it("Should be able to toggle public sharing", () => {
+    cy.visit("/admin/settings/public-sharing");
+    const publicSharingToggle = cy.findByLabelText("Enable Public Sharing");
+    publicSharingToggle.should("be.checked");
+
+    publicSharingToggle.click();
+    publicSharingToggle.should("not.be.checked");
+  });
+
+  it("should see public dashboards", () => {
+    const expectedDashboardName = "Public dashboard";
+    const expectedDashboardSlug = "public-dashboard";
+    cy.createQuestionAndDashboard({
+      dashboardDetails: {
+        name: expectedDashboardName,
+      },
+      questionDetails: {
+        name: "Question",
+        query: {
+          "source-table": ORDERS_ID,
+        },
+      },
+    })
+      .then(({ body }) => {
+        const dashboardId = body.dashboard_id;
+        cy.wrap(dashboardId).as("dashboardId");
+        cy.request("POST", `/api/dashboard/${dashboardId}/public_link`, {});
+      })
+      .then(response => {
+        cy.wrap(response.body.uuid).as("dashboardUuid");
+      });
+
+    cy.visit("/admin/settings/public-sharing");
+
+    cy.findByText("Shared Dashboards").should("be.visible");
+    cy.findByText(expectedDashboardName).should("be.visible");
+    cy.get("@dashboardUuid").then(dashboardUuid => {
+      cy.findByText(new RegExp(dashboardUuid)).should("be.visible");
+    });
+
+    cy.get("@dashboardId").then(dashboardId => {
+      cy.findByText(expectedDashboardName).should(
+        "have.attr",
+        "href",
+        `/dashboard/${dashboardId}-${expectedDashboardSlug}`,
+      );
+    });
+
+    cy.button("Revoke link").click();
+    modal().within(() => {
+      cy.findByText("Disable this link?").should("be.visible");
+      cy.button("Yes").click();
+    });
+    cy.findByText("No dashboards have been publicly shared yet.").should(
+      "be.visible",
+    );
+  });
+
+  it("should see public questions", () => {
+    const expectedQuestionName = "Public question";
+    const expectedQuestionSlug = "public-question";
+    cy.createQuestion({
+      name: expectedQuestionName,
+      query: {
+        "source-table": ORDERS_ID,
+      },
+    })
+      .then(({ body }) => {
+        const questionId = body.id;
+        cy.wrap(questionId).as("questionId");
```

Comment:
not unless you set `wrapId` = true. But I want to keep the test structure relatively the same since `createQuestionAndAddToDashboard` [doesn't wrap the dashboard ID for us.](https://github.com/metabase/metabase/blob/master/frontend/test/__support__/e2e/commands/api/dashboard.js) We might better do the same for others.